Test/git-issues/git-issue-977.dfy

```diff
@@ -177,3 +177,42 @@ lemma Test7(k: nat, n: int)
   About_NumNat(k, ToNumber(if n < 0 then 0 else n));
   assert RicochetNat(k, ToNumber(if n < 0 then 0 else n));  // info: not inlined
 }
+
+// ----- Inline/trigger issues for prefix predicates --------------------------------------------------------
+
+module PrefixBodyInlining {
+  greatest predicate AAA(r: nat)
+  {
+    BBB(r)
+  }
+
+  greatest predicate BBB(s: nat)
+
+  lemma P(k: ORDINAL, n: int)
+    requires forall t :: AAA#[k](t)
+  {
+    // We DO NOT want the following call to be inlined--a trigger would then contain a "<".
```

Comment:
Where can I find more information on this type of inlining? I don't understand this test.

Reply:
I added an explanatory comment in the test file.